src/commands/dev/types.d.ts

```diff
@@ -8,23 +8,14 @@ export type BuildConfig = NonNullable<NetlifyTOML['build']>
 export type DevConfig = NonNullable<NetlifyTOML['dev']> & {
```

Comment:
You can see here we're extending [this type](https://github.com/netlify/build/blob/1b9f2f40de7ded6285e9b7c08a1adb43550a0ccf/packages/build-info/src/settings/netlify-toml.ts#L378-L389), but we were duplicating most of it here. Most of this diff is cleaning up this unnecessary duplication + adding `undefined` to optional properties.